### PR TITLE
feat: add shell-completions subcommand

### PR DESCRIPTION
**Pull request by lamchau**
extracted from an experimental _'daemon/lsp mode'_ branch which may overhaul how session management is handled. at the very least, landing this would be a good starting point to enable discoverability of commands and adoption.

 - `shell-completions --install` support for `bash`, `fish`, or `zsh` (appends to standard config files)
 - `shell-completions --generate` support for `bash`, `fish`, or `zsh` (custom location with install text)
 - support for `goose session resume <session_name>`

this can only be tested via an entrypoint (i.e. installed not via `uv`)

```fish
pipx install . --preinstall goose-plugins-block --force
```
## Usage

This will install to the set user-defined `$SHELL`
```fish
goose shell-completions --install
```

```
Usage: goose shell-completions [OPTIONS] [[bash|zsh|fish]]

  Manage shell completions for goose

Options:
  --install   Install shell completions
  --generate  Generate shell completions
  --help      Show this message and exit.
```

Note: For `bash` and `zsh`, the user will need to `source {rc_file}` before it can take effect.

Alternatively, you can generate the completions and manually install run them yourself
```fish
goose shell-completions --generate bash > path/to/custom-bash
goose shell-completions --generate fish > path/to/completion.fish
goose shell-completions --generate zsh > path/to/custom-zsh
```

## Example installations with autocompletion per `$SHELL`
image

### Guard for exclusive options `--generate` or `--install` only
image

## Autocomplete for session (via `fish`; `goose session resume a<tab>`)
image

### CODE REVIEW

File: src/goose/utils/autocomplete.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+def setup_zsh() -> None:
+    zshrc = Path("~/.zshrc").expanduser()
+    if is_autocomplete_installed(zshrc):
+        return
+
+    with open(zshrc, "a") as f:
+        f.write("autoload -U +X compinit && compinit\n")
+        f.write("autoload -U +X bashcompinit && bashcompinit\n")
+        f.write('eval "$(_GOOSE_COMPLETE=zsh_source goose)"\n')
```
Comment (alexgorbatchev) [inline src/goose/utils/autocomplete.py:74]: IMO this shouldn't write to files. For instance my `~/.zshrc` is a bootstrap for my dotfiles, i already have `autoload compinit` elsewhere in files. My preference would be for these to be printed to STDOUT as instructions and let me do it by hand.

If I run this, I will have to go into my files, rip this out and move into the right place. I think many people do maintain their own dotfiles.

Comment (lamchau) [reply src/goose/utils/autocomplete.py:74]: easy enough! added `generate-shell-completions --shell=<shell>` in [9165e61](https://github.com/square/goose/pull/76/commits/9165e616f261347318695c4f9838a14123bfe3b4)

Comment (lamchau) [reply src/goose/utils/autocomplete.py:74]: combined both commands into one so it's now `goose shell-completions --generate`

**Review by baxen — APPROVED** [review]
LGTM

**Review by michaelneale — APPROVED** [review]

**Pull request merged**